agent/structs/config_entry_routes.go

```diff
@@ -103,10 +113,28 @@ func (e *TCPRouteConfigEntry) GetMeta() map[string]string {
 }
 
 func (e *TCPRouteConfigEntry) Normalize() error {
+	for i, parent := range e.Parents {
+		if parent.Kind == "" {
+			parent.Kind = APIGateway
+			e.Parents[i] = parent
+		}
+	}
 	return nil
 }
 
 func (e *TCPRouteConfigEntry) Validate() error {
+	validParentKinds := map[string]bool{
+		APIGateway: true,
+	}
+
+	if len(e.Services) > 1 {
+		return fmt.Errorf("tcp-based routes currently only support one service")
+	}
+	for _, parent := range e.Parents {
+		if !validParentKinds[parent.Kind] {
+			return fmt.Errorf("unsupported parent kind: %q", parent.Kind)
```

Comment:
I think we should consider including the list of acceptable parent kinds in the error message here, assuming that it will appear on the resource's status post-validation and will only ever be a handful of acceptable kinds.